### PR TITLE
feat: s3 drop and replace so it's usable for Athena

### PR DESCRIPTION
**Pull request by albertyfwu**

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -2,7 +2,6 @@ DATABASE_URL=postgresql://postgres:postgres@localhost:5432/supaglue?schema=publi
 API_HOST=
 API_KEY=
 PAGE_SIZE=1000
-PROVIDER_NAME=
```
Comment (albertyfwu) [inline .env.sample:5]: never used. we use the provider name given in the webhook

File: index.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ app.post("/supaglue_sync_webhook", async (req, res) => {
     customerId,
     providerName,
     destination: isS3Destination
-      ? new S3Destination(objectListName, syncStartTime)
+      ? new S3Destination(objectListName, customerId, providerName, syncStartTime)
```
Comment (albertyfwu) [inline index.ts:86]: we don't want to drop and replace all objectlistname/customerid/providername. only the one that actually came in

File: lib/destinations/s3.ts [hunk 3]
```diff
@@ -11,24 +11,68 @@ const client = new S3Client({
...
+  getKeyPrefix() {
+    return `${this.objectListName}/${this.customerId}/${this.providerName}`;
+  }
+
+  async dropExistingRecordsIfNecessary() {
+    console.log('Dropping existing S3 records...', {
+      objectListName: this.objectListName,
+      customerId: this.customerId,
+      providerName: this.providerName,
+    });
```
Comment (lucasmarshall) [inline lib/destinations/s3.ts:34]: Use logger instead please

File: lib/destinations/s3.ts [hunk 4]
```diff
@@ -11,24 +11,68 @@ const client = new S3Client({
...
+      }
+    );
+
+    for await (const page of paginator) {
+      const keys = page.Contents?.flatMap((content) => content.Key ?? []) ?? [];
+      console.log('Dropping S3 objects by keys:', JSON.stringify(keys), {
+        objectListName: this.objectListName,
+        customerId: this.customerId,
+        providerName: this.providerName,
+      });
```
Comment (lucasmarshall) [inline lib/destinations/s3.ts:51]: Same

File: lib/destinations/s3.ts [hunk 5]
```diff
@@ -11,24 +11,68 @@ const client = new S3Client({
...
+      await client.send(command);
+    }
   }
 
   async write(results: any[]): Promise<void> {
+    console.log('Writing S3 objects', {
+      objectListName: this.objectListName,
+      customerId: this.customerId,
+      providerName: this.providerName,
+    });
```
Comment (lucasmarshall) [inline lib/destinations/s3.ts:67]: Same

**Review by lucasmarshall — CHANGES_REQUESTED** [review]

**Review by lucasmarshall — APPROVED** [review]

**Pull request merged**